src/formula/string_utils.hpp

```diff
@@ -49,7 +49,7 @@ inline bool might_contain_variables(const std::string &str)
  * 'str' with the equivalent symbols in the given symbol table. If 'symbols'
  * is nullptr, then game event variables will be used instead.
  */
-std::string interpolate_variables_into_string(const std::string &str, const string_map * const symbols);
+std::string interpolate_variables_into_string(const std::string &str, const std::map<std::string, t_string> * const symbols);
```

Comment:
Why was this changed? Is `string_map` not visible in this file?

Reply:
Doxygen has its own source code parsing, which doesn't always exactly match what compilers will allow. So there will be cases like this where Doxygen gets confused by something that's valid C++.

Reply:
It doesn't seem right to compromise our code just to satisfy Doxygen. If we _have_ `string_map` we should use it everywhere it's needed, not expand it in arbitrary places. If that confuses Doxygen, we need to think of a way to make it _not_ confuse Doxygen.

Reply:
I'm open to suggestions. There's no way to forcibly define certain things for Doxygen though, as far as I know.

Reply:
Okay… it's defined in the other string_utils.hpp which is literally included at the top of this file. Is Doxygen getting confused because two files have the same name…?

Reply:
Might be https://github.com/doxygen/doxygen/issues/1398 , I've seen other problems with Doxygen getting confused about overloaded functions. Some have been fixed, but there's definitely some still-open longstanding issues.

I don't have a better suggestion.